crates/rustc_codegen_spirv/src/symbols.rs

```diff
@@ -596,6 +600,249 @@ fn parse_image_type(
     }))
 }
 
+fn parse_capability_list(arg: &NestedMetaItem) -> Result<Vec<Capability>, ParseAttrError> {
+    let list = match arg.meta_item_list() {
+        Some(list) => list,
+        None => {
+            return Err((
+                arg.span(),
+                "`#[spirv(capability())]` requires a list.".to_string(),
+            ))
+        }
+    };
+
+    Ok(list
+        .iter()
+        .filter_map(NestedMetaItem::ident)
+        .filter_map(|i| {
+            CAPABILITIES
+                .iter()
+                .find(|x| Symbol::intern(x.0) == i.name)
```

Comment:
This should be cached and put in a HashMap, similar to how `const BUILTINS`/`Symbols::attributes` is done - interning is relatively expensive, especially in a O(n) find loop. The array should probably also be defined alongside the other large chunks of data, to consolidate them into something more manageable.